addons/l10n_vn/migrations/16.0.2.0.2/post-migration.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, SUPERUSER_ID
+from odoo.osv import expression
+
+FIXED_ACCOUNTS_TYPE = {
+    'expense_direct_cost': ['6271', '6272', '6273', '6274', '6277', '6278'],
+}
+
+
+def _fix_accounts_type(env):
+    for correct_account_type, accounts_code in FIXED_ACCOUNTS_TYPE.items():
+        domains_per_company = []
+        for company in env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')]):
+            domain = expression.AND([
+                [('company_id', '=', company.id), ('account_type', '!=', correct_account_type)],
+                expression.OR([
+                    [('code', 'like', f'{code}%')] for code in accounts_code
+                ])
+            ])
+            domains_per_company.append(domain)
+        accounts = env['account.account'].search(expression.OR(domains_per_company))
+        accounts.write({
+            'account_type': correct_account_type
+        })
```

Comment:
I don't understand the part `('code', 'like', f'{code}%')`, as far as I can see the codes are an exact match.
Do you think this is better?
```suggestion
    companies = env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')])
    company_ids = companies.ids
    
    accounts = env['account.account'].search([
        ('company_id', 'in', company_ids),
        ('account_type', '!=', correct_account_type),
        ('code', 'in', accounts_code)
    ])
    
    accounts.write({'account_type': correct_account_type})
```

Reply:
Because I assess that the user is capable of creating a sub-account.

Reply:
I see, thank you! Well I will let the accounting team do their review :smile: